lib/core/base/cache.js

```diff
@@ -27,4 +40,35 @@ const cache = {
   }
 };
 
+/**
+ * Validates a cache key or throws an error.
+ * @param {String} key - The key to validate.
+ * @returns {Void} Void if the key is valid.
+ * @throws {TypeError} If the key is not a string.
+ */
+function validateKey(key) {
+  if (typeof key !== 'string') {
+    throw new TypeError('key must be a string, ' + typeof key + ' given');
+  }
+  if (key === '') {
+    throw new TypeError('key must not be empty');
+  }
```

Comment:
Did you know axe-core has an `assert`? Might be good to use:

```suggestion
  assert(typeof key !== 'string', 'key must be a string, ' + typeof key + ' given');
  assert(key === '', 'key must not be empty');
```